packages/controllers/src/snaps/SnapController.ts

```diff
@@ -387,6 +429,25 @@ type FeatureFlags = {
   dappsCanUpdateSnaps?: true;
 };
 
+type SemVerVersion = string;
+export type CheckSnapBlockListArg = Record<SnapId, SemVerVersion>;
```

Comment:
SnapId doesn't really globally identify a snap. A same snap could be installed from npm, localhost, and ipfs in the future.

Adding snap's code hash to the check could be useful to check for such cases.

Reply:
Being able to check for shasums is a really good feature for the reasons you describe. I think the `id@version` combination is also useful, because that will allow us to block version ranges without storing each individual shasum. I'll try to update the `_checkBlockList` function parameter to support both.

Reply:
Tracking this here: https://github.com/MetaMask/snaps-skunkworks/issues/620

We won't include it in the initial version.